### PR TITLE
Initial implementation of the SlashNext Custom Logic Apps Connector a…

### PR DESCRIPTION
**Pull request by SaadatAbid-snx**
@…nd playbook templates.

Signed-off-by: Saadat Abid <saadat.abid.2540@slashnext.net>

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by vmanojreddy** [comment]
@devikamehra  Please review the playbooks and provide your sign off. thanks!!!

**Comment by SaadatAbid-snx** [comment]
> @devikamehra Please review the playbooks and provide your sign off. thanks!!!

@devikamehra Hi, I hope that you are doing well. Just wanted to know if the PR review has started and do let us know if anything is required from our side. Thanks

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/azure_function_deploy.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
```
Comment (devikamehra) [inline Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/azure_function_deploy.json:1]: Why do we require to create a Function App here?

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+  },
+  "resources": [
+    {
+      "name": "SlashNext_azure_LinkedTemplate",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2018-05-01",
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/azure_function_deploy.json"
```
Comment (devikamehra) [inline Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:22]: Do not include linked templates here. Include the actual resource in a single ARM template. This is applicable wherever the linked template is used.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/azure_function_deploy.json (hunk 1 above)
Comment (SaadatAbid-snx) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/azure_function_deploy.json:1]: Azure function is performing following two tasks
1 Parsing URLs form raw logs which we get as input from playbook.
2 Generation of URLs and md5 mapping .

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json (hunk 2 above)
Comment (SaadatAbid-snx) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:22]: Kindly confirm this change is only required for the playbooks that are in child folders?

The main folder contains a playbook which deploys the whole package in the environment due to which we have used linked templates in it. So as per the sample solutions, my understanding is that only this arm template does not require the mentioned change set.

Comment (devikamehra) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:22]: @SaadatAbid-snx This change is required in the main deployment script also.

Comment (SaadatAbid-snx) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json]: @devikamehra updated!

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json [hunk 3]
```diff
@@ -0,0 +1,819 @@
+{
```
Comment (devikamehra) [inline Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:1]: The deployment of this playbook gives the following error:

{
    "status": "Failed",
    "error": {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
        "details": [
            {
                "code": "Conflict",
                "message": "{\r\n  \"Code\": \"Conflict\",\r\n  \"Message\": \"Website with given name Slashnext-azure-function already exists.\",\r\n  \"Target\": null,\r\n  \"Details\": [\r\n    {\r\n      \"Message\": \"Website with given name Slashnext-azure-function already exists.\"\r\n    },\r\n    {\r\n      \"Code\": \"Conflict\"\r\n    },\r\n    {\r\n      \"ErrorEntity\": {\r\n        \"ExtendedCode\": \"54001\",\r\n        \"MessageTemplate\": \"Website with given name {0} already exists.\",\r\n        \"Parameters\": [\r\n          \"Slashnext-azure-function\"\r\n        ],\r\n        \"Code\": \"Conflict\",\r\n        \"Message\": \"Website with given name Slashnext-azure-function already exists.\"\r\n      }\r\n    }\r\n  ],\r\n  \"Innererror\": null\r\n}"
            }
        ]
    }
}

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json (hunk 3 above)
Comment (SaadatAbid-snx) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:1]: This has been fixed!

Comment (devikamehra) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:1]: The issue is resolved for individual playbook deployment but the package deployment failed with following error.
{
    "status": "Failed",
    "error": {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
        "details": [
            {
                "code": "BadRequest",
                "message": "{\r\n  \"error\": {\r\n    \"code\": \"FunctionNotFound\",\r\n    \"message\": \"The function '/subscriptions/4383ac89-7cd1-48c1-8061-b0b3c5ccfd97/resourceGroups/ankisentinelrg/providers/Microsoft.Web/sites/slashnext-azure-function-ootzm/functions/processlogs' could not be found.\"\r\n  }\r\n}"
            }
        ]
    }
}

**Comment by devikamehra** [comment]
Also, 
1. how are we fetching the values of subscription id, workspace id, resource group and workspace name?
2. Why are we using HTTP request to query log analytics? We have a logic app connector to do the same job. Can we use the same?

**Comment by devikamehra** [comment]
Validation is failing for log with the following error.
{
    "status": "Failed",
    "error": {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
        "details": [
            {
                "code": "BadRequest",
                "message": "{\r\n  \"error\": {\r\n    \"code\": \"FunctionNotFound\",\r\n    \"message\": \"The function '/subscriptions/4383ac89-7cd1-48c1-8061-b0b3c5ccfd97/resourceGroups/ankisentinelrg/providers/Microsoft.Web/sites/slashnext-azure-function-ootzm/functions/processlogs' could not be found.\"\r\n  }\r\n}"
            }
        ]
    }
}

**Comment by v-spadarthi** [comment]
Logo is valid. Please feel free to merge if other components look good.

File: Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json (hunk 3 above)
Comment (SaadatAbid-snx) [reply Solutions/SlashNext/Playbooks/SlashNextWebAccessLogAssessment/deploy.json:1]: At the moment we are not able to reproduce mentioned error in our azure environment. We used the same package playbook and it deployed in our environment without any error. Can you please guide us what steps we can follow to reproduce this issue so we can provide the fix. Thanks

**Comment by SaadatAbid-snx** [comment]
> Also,
> 
>     1. How are we fetching the values of subscription id, workspace id, resource group and workspace name? If the user is expected to provide these values, can we add them as parameter in the logic app ARM Template?
> 
>     2. Why are we using HTTP request to query log analytics? We have a logic app connector to do the same job. Can we use the same?

1- Playbooks is modified to receive value from parameters for Subscription id, workspace id, resource group and workspace name.     

2- As per the document API query can return upto 64 MB  of data as compared to Log monitor connector which can return 17MB approximately and we opted for more data for analysis.

**Review by devikamehra — APPROVED** [review]

**Review by NikTripathi — APPROVED** [review]

**Pull request merged**